### PR TITLE
Support Reshape with zero dim

### PR DESCRIPTION
**Pull request by l-bat**
resolves: #18209

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under OpenCV (BSD) License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

<cut/>

```
force_builders=Custom,Custom Win,Custom Mac
build_image:Custom=ubuntu-openvino-2020.4.0:16.04
build_image:Custom Win=openvino-2020.3.0
build_image:Custom Mac=openvino-2020.3.0

test_modules:Custom=dnn,python2,python3,java
test_modules:Custom Win=dnn,python2,python3,java
test_modules:Custom Mac=dnn,python2,python3,java

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

File: modules/dnn/src/onnx/onnx_importer.cpp [hunk 1]
```diff
@@ -1344,6 +1344,10 @@ void ONNXImporter::populateNet(Net dstNet)
             if (node_proto.input_size() == 2) {
                 Mat blob = getBlob(node_proto, constBlobs, 1);
                 CV_Assert(blob.type() == CV_32SC1);
+                if (std::find(blob.ptr<int>(), blob.ptr<int>() + blob.total(), -1) == blob.ptr<int>() + blob.total())
+                {
+                    std::replace(blob.ptr<int>(), blob.ptr<int>() + blob.total(), 0, -1);
+                }
```
Comment (dkurt) [inline modules/dnn/src/onnx/onnx_importer.cpp:1350]: May I ask to comment this condition?

Comment (l-bat) [reply modules/dnn/src/onnx/onnx_importer.cpp:1350]: I'll add comment.
I can replace condition:
`if (std::find(blob.ptr<int>(), blob.ptr<int>() + blob.total(), -1) == blob.ptr<int>() + blob.total())`
to 
`if (framework_name == "keras2onnx")`
Which of the conditions is better?

Comment (dkurt) [reply modules/dnn/src/onnx/onnx_importer.cpp:1350]: It's always better to keep generic condition. But it depends on the issue we try to resolve (Keras2ONNX bug or ONNX import)

**Review by dkurt — APPROVED** [review]
:+1: Thank you!

**Pull request merged**